### PR TITLE
Add new fields to workload.proto

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Review by keithmattix — COMMENTED** [review]
This is just bringing this proto up to date with istio/istio and ztunnel right?

File: source/extensions/common/workload_discovery/discovery.proto [hunk 1]
```diff
@@ -124,10 +127,21 @@ message Workload {
   // The cluster ID that the workload instance belongs to
   string cluster_id = 18;
 
+  // The Locality defines information about where a workload is geographically deployed
+  Locality locality = 24;
+
+  NetworkMode network_mode = 25;
```
Comment (kyessenov) [inline source/extensions/common/workload_discovery/discovery.proto:133]: missing definition

**Comment by howardjohn** [comment]
> This is just bringing this proto up to date with istio/istio and ztunnel right?

Yeah without this envoy spams warnings

**Review by keithmattix — APPROVED** [review]

**Comment by keithmattix** [comment]
Approved pending tests (looks like you just need clang-format?)

**Pull request merged**